File: src/algorithms/longest-common-prefix.js
/**
 * A function that finds the longest common prefix string amongst an array
 * of strings. If there is no common prefix, returns an empty string.
 * 
 * @param {Array} strs - An array of strings
 * @returns {string} - Returns the longest common prefix amongst the strs array
 */
function longestCommonPrefix (strs) {
  if (strs.length === 0 || strs === null) {
    return '';
  }
  for (let i = 0; i < strs[0].length; i++) {
    let char = strs[0][i];
    for (let j = 1; j < strs.length; j++) {
      if (i === strs[j].length || strs[j][i] !== char) {
        return strs[0].substring(0, i);
      }
    }
  }
  return strs[0];
}

File: src/algorithms/longest-common-prefix.py
def longest_common_prefix(strs):
    """Write a function to find the longest common prefix string amongst an 
    array of strings. If there is no common prefix, return an empty string "".

    Args:
        strs (list[str]): A list of strings made up of lowercase English 
        letters.

    Returns:
        str: Returns a string of lowercase English letters.
    """
    if len(strs) == 1:
        return strs[0]
    res = ""
    length = len(strs[0])
    for i in range(length):
        char = strs[0][i]
        for j in range(1, len(strs)):
            if i >= len(strs[j]) or strs[j][i] != char:
                return res
        res += char
    return res
